common/table.js

```diff
@@ -494,34 +494,53 @@
             }
         }
 
+        function _callOnRowClick (scope, tuples, isSelected) {
+            if (scope.onRowClickBind) {
+                return scope.onRowClickBind()(tuples, isSelected);
+            } else if (scope.onRowClick) {
+                return scope.onRowClick()(tuples, isSelected);
+            }
+        }
+
+        /**
+         * Scrolls the current container to top. The container
+         * can be the modal-body or the main-container
+         */
+        function scrollToTop() {
+            $timeout(function () {
+                // if modal is open, scroll the modal to top
+                var container = angular.element(".modal-body");
+                if (container.length) {
+                    container.scrollTo(0, 0, 100);
+                }
+
+                // otherwise scroll the main-container
+                container = angular.element(".main-container");
+                if (container.length) {
+                    container.scrollTo(0, 0, 100);
+                }
+            }, 0);
+        }
+
         /**
          * Registers the callbacks for recordTable directive and it's children.
          * @param  {object} scope the scope object
          */
-        function registerTableCallbacks(scope) {
+        function registerTableCallbacks(scope, elem, attr) {
             if (!scope.vm) scope.vm = {};
 
-            var callOnRowClick = function (scope, tuples, isSelected) {
-                if (scope.onRowClickBind) {
-                    scope.onRowClickBind()(tuples, isSelected);
-                } else if (scope.onRowClick) {
-                    scope.onRowClick()(tuples, isSelected);
-                }
-            };
-
             scope.noSelect = modalBox.noSelect;
             scope.singleSelect = modalBox.singleSelectMode;
             scope.multiSelect = modalBox.multiSelectMode;
 
             scope.$root.checkReferenceURL = function (ref) {
                 var refUri = ref.isAttributeGroup ? ref.uri : ref.location.ermrestUri;
                 if (refUri.length > tableConstants.MAX_URL_LENGTH) {
-                    $timeout(function () {
-                        $window.scrollTo(0, 0);
-                    }, 0);
-                    AlertsService.addAlert('Maximum URL length reached. Cannot perform the requested action.', 'warning');
+                    AlertsService.addURLLimitAlert();
+                    scrollToTop();
                     return false;
                 }
+                AlertsService.deleteURLLimitAlert();
```

Comment:
Meaning the user could have removed one of the selections so the above conditional will return false and we no longer want the user to see that alert?

Reply:
correct. I will comment these questions that you asked.